hs_core/hydroshare/resource.py

```diff
@@ -1112,7 +1112,9 @@ def publish_resource(user, pk):
     resource = utils.get_resource_by_shortkey(pk)
     if resource.raccess.published:
         raise ValidationError("This resource is already published")
-
+    resource.raccess.review_pending = False
+    resource.raccess.immutable = False
+    resource.raccess.save()
```

Comment:
It could be helpful to create methods that better document what action is happengin when you set the review_pending and immutable flags for different scenarios.  Even just a descriptive method name would be great and could be reused without having to think about which bool needs to be set to what.

Reply:
good call thanks @sblack-usu
added that in https://github.com/hydroshare/hydroshare/pull/5307/commits/398df283d7e2990bca3ab3271706a1e826cf56c3